[RF] Test failure on i686 only with numpy 1.24.1 (pyunittests-pyroot-roofit-roodataset-numpy)

### Describe the bug

Test failure on i686 only with numpy 1.24.1.

Fails during package build in Fedora rawhide.
Build architectures are i686, x86_64, aarch64, ppc64le and s390x.
So the failure is on the only 32 bit architecture, so could be 32 bit related.

~~~
 115/1232 Test   #69: pyunittests-pyroot-roofit-roodataset-numpy ..........................***Failed    2.72 sec
test_ignoring_out_of_range (roodataset_numpy.TestRooDataSetNumpy.test_ignoring_out_of_range)
Test that rows with out-of-range values are skipped, both for ... 
RooFit v3.60 -- Developed by Wouter Verkerke and David Kirkby 
                Copyright (C) 2000-2013 NIKHEF, University of California & Stanford University
                All rights reserved, please read http://roofit.sourceforge.net/license.txt
[#0] WARNING:DataHandling -- FAIL
test_to_numpy_and_from_numpy (roodataset_numpy.TestRooDataSetNumpy.test_to_numpy_and_from_numpy)
Test exporting to numpy and then importing back a non-weighted dataset. ... RooDataSet.from_numpy(dataSetNumpy) Ignored 58 out-of-range events
ok
test_to_numpy_and_from_numpy_weighted (roodataset_numpy.TestRooDataSetNumpy.test_to_numpy_and_from_numpy_weighted)
Test exporting to numpy and then importing back a weighted dataset. ... ok
test_to_numpy_basic (roodataset_numpy.TestRooDataSetNumpy.test_to_numpy_basic)
Basic test with a real value and a category. ... ok
test_to_numpy_derived_weight (roodataset_numpy.TestRooDataSetNumpy.test_to_numpy_derived_weight)
Test if the optional computation of derived weights works. ... ok
test_to_numpy_weighted (roodataset_numpy.TestRooDataSetNumpy.test_to_numpy_weighted)
Test with a weighted dataset. ... ok
======================================================================
FAIL: test_ignoring_out_of_range (roodataset_numpy.TestRooDataSetNumpy.test_ignoring_out_of_range)
Test that rows with out-of-range values are skipped, both for
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/root-6.26.10/bindings/pyroot/pythonizations/test/roofit/roodataset_numpy.py", line 145, in test_ignoring_out_of_range
    self.assertEqual(dataset_numpy.numEntries(), n_in_range)
AssertionError: 42 != 0
----------------------------------------------------------------------
Ran 6 tests in 1.621s
FAILED (failures=1)
~~~

### Expected behavior

No failure. This test did not fail with numpy 1.23.

### To Reproduce

1. The failing test is part of the source distribution.

### Setup

1. ROOT version 6.26.10
2. Operating system Fedora Rawhide i686
3. How you obtained ROOT - package build from source.

### Additional context

Package build from source.

--- a/bindings/pyroot/pythonizations/test/roofit/roodataset_numpy.py
+++ b/bindings/pyroot/pythonizations/test/roofit/roodataset_numpy.py
@@ -135,9 +135,15 @@ class TestRooDataSetNumpy(unittest.TestCase):
         cat.defineType("minus", -1)
         cat.defineType("plus", +1)
 
-        in_x_range = (data["x"] <= x.getMax()) & (data["x"] >= x.getMin())
-        in_cat_range = (data["cat"] == -1) | (data["cat"] == +1)
-        n_in_range = np.sum(in_x_range & in_cat_range)
+        # Use manual loop because we had some problems with numpys boolean
+        # comparisions in the past (see GitHub issue #12162).
+        n_in_range = 0
+        for i in range(n_events):
+            in_x_range = data["x"][i] <= x.getMax() and data["x"][i] >= x.getMin()
+            in_cat_range = (data["cat"][i] == -1) or (data["cat"][i] == +1)
+            is_in_range = in_x_range and in_cat_range
+            if is_in_range:
+                n_in_range = n_in_range + 1
 
         dataset_numpy = ROOT.RooDataSet.from_numpy(data, {x, cat}, name="dataSetNumpy")
 

